_signal_processing/krakenSDR_signal_processor.py

```diff
@@ -239,6 +241,46 @@ def resetPeakHold(self):
         if self.spectrum_fig_type == "Single":
             self.peak_hold_spectrum = np.ones(self.spectrum_window_size) * -200
 
+    def calculate_squelch(self, sampling_freq, N):
+        def is_enabled_auto_squelch(v):
+            return v == "Auto" or (v == "Default" and self.vfo_default_squelch_mode == "Auto")
+
+        def find_nearest(array, value):
+            array = np.asarray(array)
+            idx = (np.abs(array - value)).argmin()
+            return idx, array[idx]
+
+        if self.spectrum_fig_type == "Single":
+            spectrum_index = 1
+        else:
+            spectrum_index = 2
+        sensor1_spec = self.spectrum[spectrum_index, :]
+        default_auto_db_offset = 5
```

Comment:
@redradist please move this to the `variables`

Reply:
I cannot, because variable are in _UI module and related only to web_interface. Mixing it would be wrong conceptually